Framework/Algorithms/src/SampleCorrections/MayersSampleCorrection.cpp

```diff
@@ -134,9 +131,10 @@ void MayersSampleCorrection::exec() {
     params.sigmaSc = sampleMaterial.totalScatterXSection();
     params.cylRadius = radius;
     params.cylHeight = height;
-    MayersSampleCorrectionStrategy correction(params, inX, inputWS->readY(i),
-                                              inputWS->readE(i));
-    correction.apply(outputWS->dataY(i), outputWS->dataE(i));
+
+    MayersSampleCorrectionStrategy correction(params, inputWS->histogram(i));
+    outputWS->setHistogram(i,
+                           correction.apply(std::move(inputWS->histogram(i))));
```

Comment:
Ok, leave it is a class then but please do sufficient cleanup such that it is clear what it does. Right now it is not clear what the difference is between the histogram passed into the constructor and the histogram passed into `apply`. Can't we get rid of one of them?